source/solvers/navier_stokes_base.cc

```diff
@@ -1808,7 +1850,22 @@ NavierStokesBase<dim, VectorType, DofsType>::read_checkpoint()
       this->flow_control.read(prefix);
     }
 
-  multiphysics->read_checkpoint();
+  // The average velocity profile initial condition uses the checkpoint
+  // mechanism to fetch a previous simulation solution. If an auxiliary physics
+  // was set to false in this previous simulation and is now set as true in a
+  // new simulation, the code would try to get an inexistant solution.
+  // Therefore, the multiphysics initial condition should not be set from the
+  // checkpoint.
+  if (this->simulation_parameters.initial_condition->type !=
+        Parameters::InitialConditionType::average_velocity_profile ||
+      (this->simulation_parameters.initial_condition->type ==
+         Parameters::InitialConditionType::average_velocity_profile &&
+       this->simulation_parameters.restart_parameters.restart))
```

Comment:
Ok I understand what you mean here. It's not the most elegant thing, but I see whereyou are trying to get at.
Maybe you could write it instead as != average_velocity_field || !=restart ?
This way you would not have this double if condition.